src/components/ui4/ModalUi4.tsx

```diff
@@ -0,0 +1,183 @@
+import * as React from 'react'
+import { BackHandler, Dimensions, StyleSheet, Text, TouchableOpacity, TouchableWithoutFeedback, View } from 'react-native'
+import { AirshipBridge } from 'react-native-airship'
+import { Gesture, GestureDetector, ScrollView } from 'react-native-gesture-handler'
+import { cacheStyles } from 'react-native-patina'
+import Animated, { runOnJS, useAnimatedStyle, useSharedValue, withTiming } from 'react-native-reanimated'
+import { BlurView } from 'rn-id-blurview'
+
+import { useHandler } from '../../hooks/useHandler'
+import { fixSides, mapSides, sidesToPadding } from '../../util/sides'
+import { maybeComponent } from '../hoc/maybeComponent'
+import { Theme, useTheme } from '../services/ThemeContext'
+
+export interface ModalPropsUi4<T = unknown> {
+  bridge: AirshipBridge<T>
+  children?: React.ReactNode
+
+  // Internal padding to place inside the component.
+  paddingRem?: number[] | number
+
+  // Include a scroll area:
+  scroll?: boolean
+
+  // Gives the box a border:
+  warning?: boolean
+
+  // Called when the user taps outside the modal or clicks the back button.
+  // If this is missing, the modal will not be closable.
```

Comment:
There is no way to make the modal self-dismiss, since the modal doesn't have access to the Airship promise. You *must* handle the `onCancel` callback by doing a `bridge.resolve`, or the promise will be left dangling forever. This change simply aligns the visual with this existing reality.